### PR TITLE
Co co 476 Build Test

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
This PR builds out the strapi content type and documentation for the CoCo.  
- @parulmishra There is one bit of configuration I had to change, I needed to modify permissions on the public/uploads folder in order to allow image uploads to function.  This was done through the strapi docker file.
- Most of the code is generated by strapi, so code reviewing those files is most likely redundant.   
-  The documentation needs to be reviewed. /docs/StrapiContentManager/CoCoReadMe.md
-  The CoCo component now allows saving of draft states.  A function requested in issue #498.
-  I will deploy strapi in the dev namespace so that the @mark-a-wilson  and @HeatherRemacle  can test it out. This will be live https://strapi-pr-506-xdwiwd-dev.pathfinder.gov.bc.ca/admin/

### CODE REVIEW

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Guide To CoCo fields for Digital.gov.bc.ca strapi app.
+
+## Fields that are not displayed on website
+
+### uid `required`
+
+This is used by the frontend to navigate to the coco when clicking on the card.  The uid should be automatically generated based on the CoCo Name entered
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:7]: coco should be "CoCo"

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+### Team Name `required`
+
+Name of the team that created the CoCo
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:11]: missing period

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+## Fields displayed on the CoCo card (All fields bellow this point are displayed on the CoCo Page.
+
+### Name `required`
+
+The name of the common component.
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:17]: I'd be consistent throughout this doc with either "CoCo" or "common component".

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+A component with two fields Status and Colour.  Their values are hardcoded to be:
+- Status
+-- Mature:
+-- Alpha:
+-- Beta:
+- Colour
+-- green: Actively Developed 	
+-- yellow: Maintained
+-- red: Abandonned
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:33]: spelling: Abandoned

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+This is a multiselect field that adds common cost structures to the CoCo. For example: Free, SAS, Paid Version.  To add items to this multi-select, got to the CoCoTag content type and add it.  The new tag will appear in the CoCo dropdown.
+
+### Number of Teams `required`
+
+An estimate of the number of teams that currently use this CoCo, ideally a given CoCo team will know how many people have implemented it, however if that information is not available an estimate is acceptable.
+
+### Onboarding Time `required`
+
+Select the best estimate for the time needed to implement a given coco.  (hours days, weeks, months)
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:53]: missing period at end

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+### Support Schedule `required`
+
+What schedule is the support team available.
+
+## Fields displayed on the CoCo page, but NOT displayed on CoCo Card.
+
+### CoCo image `required`
+
+A logo or image that represents the CoCo
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:63]: missing period at end

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+### CoCoWebsite `required`
+
+This is the CoCo's website it will be linked through the project image.
+
+### WhoIsUsingThis `1 required up to 3 possible`
+
+This is a tag associated with the ministries using the CoCo
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:71]: missing period

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+### Component Creation Date `required`
+
+A best estimate of when the component was created.  Currently strapi only has a date picker, if the content manager is unsure of the exact realease date the closest month is fine.
+
+### WhyShouldIUseThis `3 required`
+
+Three of these components are needed on each CoCo to advertise it's best features. It has three fields:
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:79]: "its" instead of "it's"

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+- Heading: Eye catching headline of a few words.
+- Details: Slightly longer text giving more details.
+
+### ServiceLevelSupport `1 required more possible`
+
+Details about support provided for the component.  
+
+### Additional Technical Information `1 required more possible`
+
+Directs interested parties to other technical docs.  Two fields
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:91]: missing colon after "Two fields"

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+### Component Creation Date `required`
+
+A best estimate of when the component was created.  Currently strapi only has a date picker, if the content manager is unsure of the exact realease date the closest month is fine.
+
+### WhyShouldIUseThis `3 required`
+
+Three of these components are needed on each CoCo to advertise it's best features. It has three fields:
+
+- Icon: An icon representing the motivation (e.g. a clock for time saving)
+- Heading: Eye catching headline of a few words.
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:82]: remove period at end

Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:82]: remove period at end

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+### Additional Technical Information `1 required more possible`
+
+Directs interested parties to other technical docs.  Two fields
+
+- Header: Text describing the technical info.
+- Link: A url link to the documentation.
+
+### Requirements and Restrictions `1 required more possible`
+
+Directs interested parties to other any additional documentation about requirements and restrictions.  Two fields
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:98]: missing colon after "Two fields"

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+- Header: Text describing the documentation.
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:100]: remove periods here

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+A url that links to the components 'get started' documentation.
+
+### Get started steps `optional`
+
+Optional richtext that allows content manager to list steps to getting started using markdown. 
+
+### Support Contact `1 required more possible`
+
+A required component for contacting the component's support.  It has two fields:
+- Header: a description of the contact (e.g. Support managed through githup issues)
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:114]: Change "githup" to "GitHub"

File: docs/StrapiContentManager/CoCoReadMe.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+- PhoneURLorEmail: a phone number, website, or e-email contact.
```
Comment (mark-a-wilson) [inline docs/StrapiContentManager/CoCoReadMe.md:115]: remove extra "e-"

**Review by mark-a-wilson — CHANGES_REQUESTED** [review]
I was unable to create a new CoCo: 
* Content cannot be saved until **all** required values are entered.
* Two of those required values are the **CoverImage** and **WhyShouldIUseThis** sections, each of which require images to be selected. 
* In both places, I attempted to upload images from my Desktop (max file size was 172 KB), but received an error message: “An internal server error occurred”.

**Review by mark-a-wilson — APPROVED** [review]
Submitting an Approve review here in case it's needed to cancel out my previous Request Changes review. Thanks for tidying up the README punctuation 🙂

Have not re-tested creating the CoCo in Strapi myself — relying on @singhinderdeep's review to confirm that.

**Review by parulmishra — APPROVED** [review]

**Pull request merged**